src/trading/ClaimProceeds.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/legacy_reputation/SafeMath.sol';
+
+/**
+ * @title ClaimProceeds
+ * @dev This allows users to claim their money from a market by exchanging their shares 
+ * AUDIT: Ensure that a malicious market can't subversively cause share tokens to be paid out incorrectly.
+ */
+contract ClaimProceeds is Controlled {
+	using SafeMath for uint256;
+
+	function publicClaimProceeds(IMarket market) onlyInGoodTimes public returns(bool) {
+		require(market.isFinalized());
+		require(block.timestamp > market.getFinalizationTime() + 3 days);
+
+	    var numberOfOutcomes = market.getNumberOfOutcomes();
+		for (uint8 outcome = 0; outcome < numberOfOutcomes; ++outcome) {
+	        uint256 payoutNumerator = uint256(market.getFinalWinningReportingToken().getPayoutNumerator(outcome));
+	        IShareToken shareToken = market.getShareToken(outcome);
+	        uint256 balance = 100; // shareToken.balanceOf(msg.sender);
+	        uint256 proceedAmount = balance.mul(market.getCompleteSetCostInAttotokens() / 10**18 * payoutNumerator / market.getPayoutDenominator());
+
+	        if (balance > 0) {
+	        	// TODO shareToken.destroyShares(msg.sender, balance);
```

Comment:
Because I'm still working on it. 😸  Currently any calls to `shareToken` result in a failed transaction. 🙁  Trying to determine the cause. 🛠